lib/maplibre-compose/src/androidMain/kotlin/dev/sargunv/maplibrecompose/core/source/Source.kt

```diff
@@ -1,11 +1,27 @@
 package dev.sargunv.maplibrecompose.core.source
 
+import android.text.Html
+import android.text.style.URLSpan
 import org.maplibre.android.style.sources.Source as MLNSource
 
 public actual sealed class Source {
   internal abstract val impl: MLNSource
   internal actual val id: String
     get() = impl.id
 
+  public actual val attributionLinks: List<AttributionLink>
+    get() {
+      // TODO minSdk 24 to get rid of deprecation warning
+      @Suppress("DEPRECATION") val spanned = Html.fromHtml(impl.attribution)
```

Comment:
Actually, Compose does support creating annotated strings directly since v1.7.0, see https://developer.android.com/reference/kotlin/androidx/compose/ui/text/package-summary#(androidx.compose.ui.text.AnnotatedString.Companion).fromHtml(kotlin.String,androidx.compose.ui.text.TextLinkStyles,androidx.compose.ui.text.LinkInteractionListener)

This is **currently** an Android-only API. There's no reason for it to eventually be multiplatform, though, I'd say. Jetbrains people just need to implement an own simple multiplatform HTML parser (, which is actually not that much work).

For my project, I had to do this (because I was using Compose 1.6 but also because at some other place I take apart an HTML and split it into sections):

- **Parser**: https://github.com/streetcomplete/StreetComplete/tree/master/app/src/main/java/de/westnordost/streetcomplete/util/html
- **Html nodes to Annotated string**: https://github.com/streetcomplete/StreetComplete/blob/bf52d5ff17d3c500257e72bf1ee89ed5e17c508a/app/src/main/java/de/westnordost/streetcomplete/ui/util/Html.kt

Without tests and given a helper class for iterating through a string, that's just about 2x 170 lines of code.

I'd absolutely be happy to change the license to contribute that upstream to Jetbrains, but I don't really know where I can do that.

Anyway, I do realize this is a bit off topic, because the MapLibre SDK for iOS already provides the parsed attribution links, so parsing HTML is only necessary for Android (at the moment, i.e. disregarding supporting  JS / Desktop in the future). Although, of course, this is also relevant for when/if eventually migrating to use a TBD C Api directly instead of Android/iOS SDKs.

Reply:
Yup, I tried exactly that and then found it's Android only. 

Ideally I'd like to provide the raw attribution string here in core, and let the compose layer parse the HTML to an annotated string. Probably something to leave for when migrating to wrapping the native SDK directly. 

For now, I matched the iOS functionality that parses out attribution (minus text outside links, because the Android SDK seems to ignore it anyway and the iOS SDK handles it poorly)